src/bika/lims/content/referencesample.py

```diff
@@ -199,8 +215,16 @@ class ReferenceSample(BaseFolder):
     _at_rename_after_creation = True
 
     def _renameAfterCreation(self, check_auto_id=False):
-        from senaite.core.idserver import renameAfterCreation
-        renameAfterCreation(self)
+        ref_id = self.getManualId()
+        parent = api.get_parent(self)
+        if not api.is_valid_id(ref_id, container=parent):
```

Comment:
Maybe a validator would make more sense instead of just falling back to the ID Server, which gives the user no clear indicator why it did not work.

Reply:
Makes sense, raising a `ValueError` now with 21e7970d9c9a711426c7b69547510dc7fcf4e99c